app/api/endpoint.py

```
@@ -0,0 +1,63 @@
+        save_to_db(engine, recommend_df)
+
+        # 7. 응답 반환
+        return {
+            "user_id": user_id,
+            "recommendations": recommend_df[["brand_id", "score", "rank"]].to_dict(orient="records")
+        }
+
+    except Exception as e:
+        import traceback
+        print("[ERROR]", traceback.format_exc())
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)